### PR TITLE
Added userlike provider

### PR DESCRIPTION
**Pull request by gazpachu**
I've installed the package like this: `npm install --save ../react-live-chat-loader` (no errors)

Then imported it with `import { LiveChatLoaderProvider, Userlike } from 'react-live-chat-loader'`

Added the Provider with `<LiveChatLoaderProvider providerKey="xxxx" provider="userlike">`

And called the widget with `<Userlike />`

I get this error trying to test it on my Next.js website:

Screenshot 2020-12-31 at 12 11 29

```
index.tsx?8abf:304 Uncaught     at resolveDispatcher (file:///.../Dev/react-live-chat-loader/node_modules/react/cjs/react.development.js:1465:13)
    at useState (file:///.../Dev/react-live-chat-loader/node_modules/react/cjs/react.development.js:1496:20)
    at LiveChatLoaderProvider (file:///.../Dev/react-live-chat-loader/dist/index.js:108:39)
    at processChild (file:///.../node_modules/react-dom/cjs/react-dom-server.node.development.js:3353:14)
    at resolve (file:///.../node_modules/react-dom/cjs/react-dom-server.node.development.js:3270:5)
    at ReactDOMServerRenderer.render (file:///.../node_modules/react-dom/cjs/react-dom-server.node.development.js:3753:22)
    at ReactDOMServerRenderer.read (file:///.../node_modules/react-dom/cjs/react-dom-server.node.development.js:3690:29)
    at renderToString (file:///.../node_modules/react-dom/cjs/react-dom-server.node.development.js:4298:27)
    at Object.renderPage (file:///.../node_modules/next/dist/next-server/server/render.js:50:851)
    at Function.getInitialProps (webpack-internal:///./node_modules/next/dist/pages/_document.js:141:19)
```

### CODE REVIEW

**Comment by mikedijkstra** [comment]
Hi @gazpachu,

We recommend that you install the package from npm rather than referencing it locally (e.g. `npm install react-live-chat-loader`).

When you reference it locally you need to link React so it uses the same version of React. You can find out more [here in the React documentation](https://reactjs.org/warnings/invalid-hook-call-warning.html#duplicate-react).

Also, I believe this would be better as an [Issue](https://github.com/calibreapp/react-live-chat-loader/issues) rather than a Pull Request so I'm going to close this for now.

Please open a new issue if the above doesn't solve your problem.

Thanks!

**Comment by mikedijkstra** [comment]
Hi @gazpachu,

I've just had another look at this and can see you're in the process of adding a new integration.

To test locally you'll need to link React with so you're using the same React in your Next.JS website and the live chat loader. [See this link for more information](https://reactjs.org/warnings/invalid-hook-call-warning.html#duplicate-react).

**Comment by benschwarz** [comment]
Hey @gazpachu, just wanted to see if there's anything we can to do help finalise this?

**Comment by gazpachu** [comment]
Hi @benschwarz, I've been quite busy at work. Will try to finish it this weekend

**Comment by gazpachu** [comment]
> you'll need to link React
I'm not sure how to do that link. I've tried using the same react version on both package.json but it didn't fix it.

**Comment by gazpachu** [comment]
Ok, ready to review

**Review by mikedijkstra — CHANGES_REQUESTED** [review]
Thanks for the PR @gazpachu!

After testing it out I've made a few suggestions for improvements and spotted one bug.

I also noticed in the Userlike UI that there are some additional configuration options, such as:

- Button position
- Button offset
- Button style (square or round)
- Button profile desktop
- Button profile mobile

I think it'd be good to include these, is this something you're happy to add to this PR? I imagine the button position would be like the `horizontalPosition` prop on the `HelpScout` widget and the others we could add as additional props to the `Userlike` component.

File: src/components/Userlike/index.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+  if (state === STATES.COMPLETE) return null
+  return (
+    <div style={{ ...styles.container }}>
+      <button
+        onClick={() => loadChat({ open: true })}
+        onMouseEnter={() => loadChat({ open: false })}
+        style={{
+          ...styles.button,
+          color,
```
[the remark this answers is not in the document]
Comment (gazpachu) [reply src/components/Userlike/index.js:50]: I was using `fill="currentColor"`, but yeah, we can apply the color directly to the SVG instead

**Comment by gazpachu** [comment]
Thanks for the review!

**Comment by gazpachu** [comment]
Regarding these two options:

- Button profile desktop
- Button profile mobile

I think from the plugin perspective, we only need to know if the fake button needs to be shown or not, on either desktop or mobile. For that case, I'd prefer to leave that logic outside this plugin.

**Comment by mikedijkstra** [comment]
> I think from the plugin perspective, we only need to know if the fake button needs to be shown or not, on either desktop or mobile. For that case, I'd prefer to leave that logic outside this plugin.

Okay, I'm happy to leave that out too then.

I think the last thing is make the `provider.open()` call work. Because `userlike` isn't immediately available we have the same problem as before. The idea behind this function is that if a user clicks on the faux chat widget then we open it up after we've loaded the real chat widget. Right now `userlike` isn't available so it's not loading. Do you think we could do another attempt loop to open it when available?

**Comment by gazpachu** [comment]
Now it works 👍

**Comment by benschwarz** [comment]
Hey @gazpachu, just wanted to let you know that @mikedijkstra is on leave so there will be some delay in finalising this PR.

Again, just wanted to re-iterate: Thanks for working on the Userlike provider!

**Review by mikedijkstra — APPROVED** [review]

**Comment by mikedijkstra** [comment]
@gazpachu I'm back! Thanks for your patience. The changes look good. Once you sign the CLA we can merge this in and push out a new release 🎉

**Comment by gazpachu** [comment]
> Once you sign the CLA
Done

**Comment by mikedijkstra** [comment]
@all-contributors please add @gazpachu for code

**Pull request merged**